unstructured/ingest/v2/processes/connectors/chroma.py

```diff
@@ -0,0 +1,284 @@
+                date_created
+            ).strftime(
+                "%Y-%m-%dT%H:%M:%S.%fZ",
+            )
+
+        if date_modified := data.get("metadata", {}).get("data_source", {}).get("date_modified"):
+            data["metadata"]["data_source"]["date_modified"] = cls.parse_date_string(
+                date_modified
+            ).strftime(
+                "%Y-%m-%dT%H:%M:%S.%fZ",
+            )
+
+        if date_processed := data.get("metadata", {}).get("data_source", {}).get("date_processed"):
+            data["metadata"]["data_source"]["date_processed"] = cls.parse_date_string(
+                date_processed
+            ).strftime(
+                "%Y-%m-%dT%H:%M:%S.%fZ",
+            )
+
+        if last_modified := data.get("metadata", {}).get("last_modified"):
+            data["metadata"]["last_modified"] = cls.parse_date_string(last_modified).strftime(
+                "%Y-%m-%dT%H:%M:%S.%fZ",
+            )
+
+        # String casting
+        if version := data.get("metadata", {}).get("data_source", {}).get("version"):
+            data["metadata"]["data_source"]["version"] = str(version)
+
+        if page_number := data.get("metadata", {}).get("page_number"):
+            data["metadata"]["page_number"] = str(page_number)
+
+        if regex_metadata := data.get("metadata", {}).get("regex_metadata"):
+            data["metadata"]["regex_metadata"] = str(json.dumps(regex_metadata))
+
+    @classmethod
+    def normalize_dict(cls, data: dict) -> dict:
+        """
+        Prepares dictionary in the format that Chroma requires
+        """
+        element_id = data.get("element_id", str(uuid.uuid4()))
+        return {
+            "id": element_id,
+            "embedding": data.pop("embeddings", None),
+            "document": data.pop("text", None),
+            "metadata": flatten_dict(data, separator="-", flatten_lists=True, remove_none=True),
+        }
+
+    def run(
+        self,
+        elements_filepath: Path,
+        file_data: FileData,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        with open(elements_filepath) as elements_file:
+            elements_contents = json.load(elements_file)
+        for element in elements_contents:
+            self.conform_dict(data=element)
+        normalized_elements = []
+        for element in elements_contents:
+            normalized_elements.append(self.normalize_dict(data=element))
+
+        output_path = Path(output_dir) / Path(f"{output_filename}.json")
+        with open(output_path, "w") as output_file:
+            json.dump(normalized_elements, output_file)
+        return output_path
+
+
+@dataclass
+class ChromaUploaderConfig(UploaderConfig):
+    batch_size: int = 100
+
+
+@dataclass
+class ChromaUploader(Uploader):
+    upload_config: ChromaUploaderConfig
+    connection_config: ChromaConnectionConfig
+    _collection: Optional["ChromaCollection"] = None
```

Comment:
Why are we creating a collection as part of this process? Is a collection like a table in the DB sense? If so, we should assume a user already has this before running the upload process.

Reply:
I moved things around. Its just Client here now. 

Still allowing people to create an adhoc collection though upon upload. 

I think it would be breaking behaviour to require people to have a collection already made. Chroma allows both options.